api/src/migrations/main/1718643595423-simplify-spatial-geometry.js

```diff
@@ -0,0 +1,26 @@
+const { MigrationInterface, QueryRunner } = require('typeorm');
+
+module.exports = class SimplifySpatialGeometry1718643595423 {
+    async up(queryRunner) {
+        console.log(
+            'Starting geometry simplification (update geometry column in cut_block, road_section and retention_area table) migration'
+        );
+        await queryRunner.query(`
+    
+            -- update geometry column to apply the simplification algorithm in cut_block table
+            UPDATE app_fom.cut_block SET geometry=ST_SimplifyPreserveTopology(geometry, 2.5);
+
+            -- update geometry column to apply the simplification algorithm in retention_area table
+            UPDATE app_fom.retention_area SET geometry=ST_SimplifyPreserveTopology(geometry, 2.5);
+
+            -- update geometry column to apply the simplification algorithm in road_section table
+            UPDATE app_fom.road_section SET geometry=ST_SimplifyPreserveTopology(geometry, 2.5);
+        `);
+    }
```

Comment:
This is good, but just a question.
Is it possible to loop for each record and update so that we know which one is (or how many are) successfully updated but others are failed? Or is it all or none kind of update?
@basilv

Reply:
These bulk updates are fine, we're not concerned if in some cases the geometry is already simple enough and doesn't need to be changed.